src/main/java/org/folio/reshare/index/storage/Storage.java

```diff
@@ -50,23 +61,62 @@ public Storage(Vertx vertx, String tenant) {
                 + " (local_identifier, library_id)",
-            "CREATE INDEX IF NOT EXISTS idx_bib_record_match_key ON " + bibRecordTable
-                + " (match_key)",
             "CREATE OR REPLACE VIEW " + itemView
-                + " AS SELECT id, local_identifier, library_id, match_key,"
+                + " AS SELECT id, local_identifier, library_id,"
                 + " jsonb_array_elements("
                 + " (jsonb_array_elements((inventory->>'holdingsRecords')::JSONB)->>'items')::JSONB"
-                + ") item FROM " + bibRecordTable
+                + ") item FROM " + bibRecordTable,
+            "CREATE TABLE IF NOT EXISTS " + matchTypeTable
+                + "(id uuid NOT NULL PRIMARY KEY,"
+                + " code VARCHAR, "
+                + " name VARCHAR)",
+            "CREATE UNIQUE INDEX IF NOT EXISTS idx_match_type_code ON " + matchTypeTable
+                + " (code)",
+            "INSERT INTO " + matchTypeTable
+                + " (id, code, name) "
+                + " VALUES ('" + MATCH_TYPE_ID_ISBN + "', 'ISBN', 'ISBN') ON CONFLICT DO NOTHING",
+            "CREATE TABLE IF NOT EXISTS " + matchPointTable
+                + "(bib_record_id uuid NOT NULL,"
+                + " match_type_id uuid NOT NULL,"
+                + " match_value VARCHAR NOT NULL,"
+                + " CONSTRAINT match_point_fk_bib_record FOREIGN KEY "
+                + "                (bib_record_id) REFERENCES " + bibRecordTable
+                + " ON DELETE CASCADE,"
+                + " CONSTRAINT match_point_fk_type FOREIGN KEY "
+                + "                (match_type_id) REFERENCES " + matchTypeTable + ")",
+            "CREATE UNIQUE INDEX IF NOT EXISTS match_point_idx ON " + matchPointTable
+                + " (match_type_id, match_value, bib_record_id)",
+            "CREATE INDEX IF NOT EXISTS match_point_bib_id_idX ON " + matchPointTable
+                + " (bib_record_id)",
+            "CREATE OR REPLACE FUNCTION " + matchPointFunction
+                + " RETURNS TRIGGER AS $match_points$"
+                + " BEGIN"
+                + "   IF (TG_OP = 'UPDATE') THEN"
+                + "     DELETE FROM " + matchPointTable
+                + "     WHERE bib_record_id = NEW.id;"
+                + "   END IF;"
+                + "   IF (TG_OP = 'UPDATE' OR TG_OP = 'INSERT') THEN"
+                + "     INSERT INTO " + matchPointTable
+                + "       (bib_record_id, match_type_id, match_value)"
+                + "     SELECT NEW.id, '" + MATCH_TYPE_ID_ISBN + "', identifier->>'value'"
+                + "     FROM jsonb_array_elements("
+                + "           ((NEW.inventory->>'instance')::JSONB->>'identifiers')::JSONB) "
+                + "            AS identifier"
+                + "            WHERE identifier->>'identifierTypeId'='"
+                +                    IDENTIFIER_TYPE_ID_ISBN + "';"
+                + "   END IF;"
```

Comment:
I'm not much fund of triggers or SQL functions.. 1: I don't understand it (:-) .. 2: They are difficult to unit test.. 3. I think parsing of JSON should be done in Java code and knowledge about UUIDs in inventory should definitely not live in an SQL function.. 4: upgrading problems with triggers/functions.. 

So if the logic is "universal", a trigger MAY be good.. But I don't think in this case, that applies..